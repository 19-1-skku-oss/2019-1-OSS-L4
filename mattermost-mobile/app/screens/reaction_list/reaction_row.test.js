// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import ReactionRow from './reaction_row';

describe('ReactionRow', () => {
    const baseProps = {
        emojiName: 'smile',
        navigator: {},
        teammateNameDisplay: 'username',
        theme: Preferences.THEMES.default,
        user: {id: 'user_id', username: 'username'},
    };

    test('should match snapshot, renderContent', () => {
        const wrapper = shallow(
            <ReactionRow {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
