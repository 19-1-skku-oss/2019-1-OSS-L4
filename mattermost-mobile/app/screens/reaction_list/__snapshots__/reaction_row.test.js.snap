// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactionRow should match snapshot, renderContent 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "flex-start",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "width": "13%",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "paddingTop": 3,
          }
        }
      >
        <Connect(ProfilePicture)
          showStatus={false}
          size={24}
          userId="user_id"
        />
      </View>
    </TouchableOpacity>
  </View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Object {
        "flexDirection": "row",
        "width": "74%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 14,
          "paddingRight": 5,
        }
      }
    >
      @username
    </Text>
    <Text>
        
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(0,0,0,0.5)",
          "fontSize": 14,
        }
      }
    >
      username
    </Text>
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "13%",
      }
    }
  >
    <Connect(Emoji)
      emojiName="smile"
      size={24}
    />
  </View>
</View>
`;
